src/wasm.h

```diff
@@ -1441,6 +1441,8 @@ class Module {
   std::map<Name, Import*> importsMap;
   std::map<Name, Export*> exportsMap;
   std::map<Name, Function*> functionsMap;
+  // Types of functions which are declared but not defined.
+  std::map<Name, FunctionType*> externTypesMap;
```

Comment:
OK, I moved the mapping to the linker but decided to let the wasm module continue to own the function type itself, to allow for deduping and because it already has the unique_ptrs.